File: apps/mobile-app/ios/VaultModels/VaultMetadata.swift
import Foundation

public struct VaultMetadata: Codable {
    public var publicEmailDomains: [String]?
    public var privateEmailDomains: [String]?
    public var vaultRevisionNumber: Int
    
    public init(publicEmailDomains: [String]? = nil, privateEmailDomains: [String]? = nil, vaultRevisionNumber: Int) {
        self.publicEmailDomains = publicEmailDomains
        self.privateEmailDomains = privateEmailDomains
        self.vaultRevisionNumber = vaultRevisionNumber
    }
}

File: apps/mobile-app/android/app/src/main/java/net/aliasvault/app/vaultstore/models/VaultMetadata.kt
package net.aliasvault.app.vaultstore.models

/**
 * Vault metadata object.
 */
data class VaultMetadata(
    /**
     * The public email domains of the vault.
     */
    val publicEmailDomains: List<String> = emptyList(),

    /**
     * The private email domains of the vault.
     */
    val privateEmailDomains: List<String> = emptyList(),

    /**
     * The revision number of the vault.
     */
    val vaultRevisionNumber: Int = 0,
)
